nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/ProcessorSummaryStatusEndpointMerger.java

```diff
@@ -0,0 +1,100 @@
+
+    @Override
+    public boolean canHandle(final URI uri, final String method) {
+        return "POST".equalsIgnoreCase(method) && SCHEDULE_SUMMARY_URI.equals(uri.getPath());
+    }
+
+    @Override
+    public NodeResponse merge(final URI uri, final String method, final Set<NodeResponse> successfulResponses, final Set<NodeResponse> problematicResponses, final NodeResponse clientResponse) {
+        if (!canHandle(uri, method)) {
+            throw new IllegalArgumentException("Cannot use Endpoint Mapper of type " + getClass().getSimpleName() + " to map responses for URI " + uri + ", HTTP Method " + method);
+        }
+
+        final ProcessorScheduleSummariesEntity responseEntity = clientResponse.getClientResponse().readEntity(ProcessorScheduleSummariesEntity.class);
+
+        // Create mapping of Processor ID to its schedule Summary.
+        final Map<String, ProcessorScheduleSummaryEntity> scheduleSummaries = responseEntity.getScheduleSummaries().stream()
+            .collect(Collectors.toMap(entity -> entity.getScheduleSummary().getId(), entity -> entity));
+
+        for (final NodeResponse nodeResponse : successfulResponses) {
+            final ProcessorScheduleSummariesEntity nodeResponseEntity = nodeResponse == clientResponse ? responseEntity :
+                nodeResponse.getClientResponse().readEntity(ProcessorScheduleSummariesEntity.class);
+
+            for (final ProcessorScheduleSummaryEntity processorEntity : nodeResponseEntity.getScheduleSummaries()) {
+                final String processorId = processorEntity.getScheduleSummary().getId();
+
+                final ProcessorScheduleSummaryEntity mergedEntity = scheduleSummaries.computeIfAbsent(processorId, id -> new ProcessorScheduleSummaryEntity());
+                merge(mergedEntity, processorEntity);
+            }
+        }
+
+        final ProcessorScheduleSummariesEntity mergedEntity = new ProcessorScheduleSummariesEntity();
+        mergedEntity.setScheduleSummaries(new ArrayList<>(scheduleSummaries.values()));
+        return new NodeResponse(clientResponse, mergedEntity);
+    }
+
+    private void merge(final ProcessorScheduleSummaryEntity target, final ProcessorScheduleSummaryEntity additional) {
+        PermissionsDtoMerger.mergePermissions(target.getPermissions(), additional.getPermissions());
+
+        final ProcessorScheduleSummaryDTO targetSummaryDto = target.getScheduleSummary();
+        final ProcessorScheduleSummaryDTO additionalSummaryDto = additional.getScheduleSummary();
+
+        // If name is null, it's because of permissions, so we want to nullify it in the target.
+        if (additionalSummaryDto.getName() == null) {
+            targetSummaryDto.setName(null);
+        }
+
+        targetSummaryDto.setActiveThreadCount(targetSummaryDto.getActiveThreadCount() + additionalSummaryDto.getActiveThreadCount());
+
+        final String additionalRunStatus = additionalSummaryDto.getRunStatus();
```

Comment:
Yes. If any node indicates that the Processor is running, it is to be considered Running. Otherwise, if any node considers it invalid, it should be invalid. Otherwise, if any node indicates it is Validating, it should be Validating because its validation is not yet known. Invalid vs. Validating I would say is debatable. If one node says Validating and the other Invalid, we could say it is Invalid or Validating because the validation is not actually known. On the other hand, regardless of whether Validating becomes Valid or Invalid, the overall status will be Invalid if any is Invalid. So the precedence there is kind of irrelevant.

Reply:
My issue is that the outcome depends on which order we process the nodes.
So considering `nodeRunStatuses -> mergedRunStatus`, then we have these outcomes:
```source
(running, invalid) -> invalid
(invalid, running) -> running
```
etc.

Reply:
Wow, yes. Thanks. Now I understand. I think we have this same bug elsewhere then :) Will address.

Reply:
Looking at how we merge the RunStatus in StatusMerger, I see the confusion/issue. We should only be checking for states of Invalid / Validating. If the additional run status is Validating or Invalid, then set the target's RunStatus to that. Otherwise, leave it as is. That way, if any node has a status of Validating/Invalid, then that will take precedence. Which one of those wins doesn't really matter, as mentioned above. This is the logic used elsewhere, so will update this PR to use that logic as well.